framework/include/geomsearch/PenetrationLocator.h

```diff
@@ -102,6 +102,9 @@ class PenetrationLocator : Restartable
   Real _normal_smoothing_distance; // Distance from edge (in parametric coords) within which to
                                    // perform normal smoothing
   NORMAL_SMOOTHING_METHOD _normal_smoothing_method;
+
+  bool _automatic_patch_update; // Update patch for the nodes saved in recheck_slave_nodes for
```

Comment:
This name (along with its associated getter method) sounds really similar to the existing `patch_update_strategy = auto` option, and the two might be confused. Maybe something like `_recheck_out_of_patch`?

Reply:
Also, make it `const`.

Reply:
Thanks, I will make this change.

Reply:
How about `recheck_patch_nonlinear_iter` instead of `automatic_patch_update`?

Reply:
Sure, that works.  I just had another thought -- I wonder if it would make any sense to have this just be another option in the `patch_update_strategy` enum. We probably don't want to change the meaning of `auto`.

Reply:
Well I can add it as an option (nonlinear_iter) to `patch_update_strategy`. But then we cannot use `auto` patch update strategy and along with this strategy of updating patch at nonlinear iterations. Till we have the dynamic Jacobian allocation, it might be better to use the auto patch update strategy along with this patch updating at nonlinear iterations to make sure Jacobian pre-allocation penalties are not too high.

Reply:
I think we shouldn't worry about the Jacobian pre-allocation penalties. If we actually pick up interactions that weren't detected before, it would be an error to not enforce them. I think we should either try to enforce them (even it it's at a high cost), or error out. Maybe it would be useful to provide a report on how many of these we actually found in an analysis, which would give users information needed to maybe change the patch size or update strategy.

Reply:
That makes sense. I will leave it on by default. I will add another option to patch_update_strategy named `nonlinear_iter` which will turn this search on. If the user uses `never` or `always` or `auto` and if are any slave nodes are not in contact with the master boundary, it will show a warning asking the user to switch to `nonlinear_iter` strategy if they expect all slave nodes to be in contact with master boundary.

I will run the fuel problems with different patch update strategies to see how often the patch updates gets called.